pymongo/command_cursor.py

```diff
@@ -386,13 +388,13 @@ def _unpack_response(
         codec_options: CodecOptions,
         user_fields: Optional[Mapping[str, Any]] = None,
         legacy_response: bool = False,
-    ) -> List[Mapping[str, Any]]:
+    ) -> List[_DocumentOut]:
         raw_response = response.raw_response(cursor_id, user_fields=user_fields)
         if not legacy_response:
             # OP_MSG returns firstBatch/nextBatch documents as a BSON array
             # Re-assemble the array of documents into a document stream
             _convert_raw_document_lists_to_streams(raw_response[0])
-        return raw_response
+        return cast(List[_DocumentOut], raw_response)
```

Comment:
We don't need this cast. Instead let's say this returns Mapping and add a # type ignore. The RawBatch classes will never be compatible with the super class because they yield raw bytes, not decoded documents.